File: Quicksort.cpp
#include<bits/stdc++.h>
using namespace std;
int  Partition(int l,int h,int arr[] ){
    int i=l-1;
    int pivot=arr[h];
    int j;
    for(j=l;j<h;j++){
        if(arr[j]<pivot){
            i++;
            int temp=arr[i];
            arr[i]=arr[j];
            arr[j]=temp;
        }
    }
    i++;
    int temp=arr[i];
    arr[i]=pivot;
    arr[h]=temp;
    return i;
}
int Quicksort(int arr[],int l,int h){
    if(l<h){
       int pivot=Partition(l,h,arr);
       Quicksort(arr,l,pivot-1);
       Quicksort(arr,pivot+1,h);
    }
}

int main()
{
    int n;
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
   int l=0;
   int h=n-1;
   Quicksort(arr,l,h);
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";

    }
  return 0;
}

File: Quicksort.java
import java.util.Arrays;
import java.util.Scanner;

public class Quicksort {

    static int partition(int l,int h,int []arr){
        int i=l-1;
        int pivot=arr[h];
        for(int j=l;j<h;j++){
            if(arr[j]<pivot){
                i++;
                int temp=arr[i];
                arr[i]=arr[j];
                arr[j]=temp;
            }
        }
        i++;
        int temp=arr[i];
        arr[i]=pivot;
        arr[h]=temp;

        return i;
    }
    static void quickAlgo(int l,int h, int []arr){
        if(l<h){
            int pivot=partition(l, h, arr);
            quickAlgo(l, pivot-1, arr);
            quickAlgo(pivot+1, h, arr);
        }

    }
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        int []arr=new int[n];
        for(int i=0;i<n;i++){
            arr[i]=sc.nextInt();
        }
        quickAlgo(0, n-1, arr);
        System.out.println(Arrays.toString(arr));
    }
    
}
